Added postgis to Vagrant script and added API test

mbu13 opened the pull request.

SillyInventor commented:
1. Move this PR to RPI-HASS/rpi_csdt_community
2. Fix indentation on line 38
3. The preferred method for getting a URL in a test is  url = reverse('api-gis') //you may need to name it
4. The preferred method for checking a response is with:
```
from rest_framework.test import APIClient
self.client = APIClient()
response = self.client.get(url)
```
5. Run python manage.py test often to see if it's working well.

chuck211991 commented:
Hey Matt, thanks for the code!

We recently moved from this repo to one located in another organization because of legal reasons. The new one is located (https://github.com/RPI-HASS/rpi_csdt_community). I don't know if there's an easy way to create a pull request to that one without deleting and reforking the repository.

For the test itself, would you mind using self.client instead of urllib2? Sometimes Django does magic with the test client to run tests in parallel, and it might get mixed up if we're using urllib2.

Thanks again!

mbu13 commented:
Thanks, I moved the PR to the HASS repo and made the necessary changes. I added a name for the url so we could test it with reverse(), but from what I noticed, the url is contingent on the ENABLE_GIS flag to be set to true. I tried changing that setting in the settings.py file but lead to other issues in the settings, so I left the flag as is. I currently named the '^api-gis/' url, but let me know if that is not the right one.
